src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/iOS/WebUI.cs

```diff
@@ -44,6 +49,30 @@ public WebUI(IPlatformParameters parameters)
             {
                 throw new ArgumentException("parameters should be of type PlatformParameters", "parameters");
             }
+
+            DidEnterBackgroundNotification = NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.DidEnterBackgroundNotification, OnMoveToBackground);
+            WillEnterForegroundNotification = NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.WillEnterForegroundNotification, OnMoveToForeground);
+        }
+
+        void OnMoveToBackground(NSNotification notification)
+        {
+            taskId = UIApplication.SharedApplication.BeginBackgroundTask(() => {
+                inBackground = false;
+                UIApplication.SharedApplication.EndBackgroundTask(taskId);
+            });
+
+            //After iOS 11.3, it is neccesary to keep a background task running while moving an app to the background in order to prevent the system from reclaiming network resources from the app. 
+            //This will prevent authentication from failing while the application is moved to the background while waiting for MFA to finish.
+            new Task(() => {
```

Comment:
what does the Task do? I see you're already doing BeginBackgroundTask, so why is this necessary?

Reply:
@oldalton The BeginBackgroundTask does not actually start the task. it simply notifies the system that the following code should not be terminated. it also registers a task id for it. The space in between BeginBackgroundTask and EndBackgroundTask is where the task is created and executed.

Reply:
Yes, I know. But what does that Task do? If I'm reading the code correctly, it just starts a while loop and waits for the boolean to change. So, if it doesn't do anything, why is it necessary?